### PR TITLE
feat: document the new `split` option

### PR DESCRIPTION
**Pull request by ematipico**
#### What kind of changes does this PR include?

- New or updated content

#### Description

Reference PR https://github.com/withastro/astro/pull/7220

In this PR we add a new way to bundle the SSR code. This PR is meant to add a new guide in the `server-side-rendering` 
page.

Also, an integration hook has been updated. This PR updates it.

Closes https://github.com/withastro/docs/issues/3503

### CODE REVIEW

File: src/content/docs/en/guides/server-side-rendering.mdx [hunk 1]
```diff
@@ -366,3 +366,33 @@ const factPromise = fetch('https://catfact.ninja/fact')
 ```
 
 In this example, `A name` will render while `personPromise` is loading. At that point, `A fact` will be appear, while `factPromise` is loading.
+
+
+#### Split the SSR build into multiple files
+
+By default, Astro builds all pages into one single file called `entry.mjs`. With the `split` option, you tell Astro emit
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/server-side-rendering.mdx:373]: Small typo!
```suggestion
By default, Astro builds all pages into one single file called `entry.mjs`. With the `split` option, you tell Astro to emit
```

**Review by Yan-Thomas — COMMENTED** [review]
Added a small suggestion, besides this LGTM, great work 🙌

File: src/content/docs/en/guides/server-side-rendering.mdx [hunk 2]
```diff
@@ -366,3 +366,33 @@ const factPromise = fetch('https://catfact.ninja/fact')
...
+multiple files, where the number of files is the number of pages:
```
Comment (ElianCodes) [inline src/content/docs/en/guides/server-side-rendering.mdx:374]: These also should be one line I think
```suggestion
By default, Astro builds all pages into one single file called `entry.mjs`. With the `split` option, you tell Astro to emit multiple files, where the number of files is the number of pages:
```

File: src/content/docs/en/guides/server-side-rendering.mdx [hunk 3]
```diff
@@ -366,3 +366,33 @@ const factPromise = fetch('https://catfact.ninja/fact')
...
+```js ins={6}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  build: {
+      split: true
+  }
+});
+```
```
Comment (ElianCodes) [inline src/content/docs/en/guides/server-side-rendering.mdx:385]:
```suggestion
```js title="astro.config.mjs" ins={6}
import { defineConfig } from 'astro/config';

export default defineConfig({
  build: {
      split: true
  }
});
```
```

File: src/content/docs/en/guides/server-side-rendering.mdx [hunk 4]
```diff
@@ -366,3 +366,33 @@ const factPromise = fetch('https://catfact.ninja/fact')
...
+
+Inside the `pages/` directory, the pages will look like this:
+
+```plaintext
+├── pages
+│   ├── blog
+│   │   ├── entry._slug_.astro.mjs
+│   │   └── entry.about.astro.mjs
+│   └── entry.index.astro.mjs
+```
```
Comment (ElianCodes) [inline src/content/docs/en/guides/server-side-rendering.mdx:398]: This should be changed with the `FileTree` component

**Review by ElianCodes — COMMENTED** [review]
Looks good, left a couple suggestions for you to review

File: src/content/docs/en/guides/server-side-rendering.mdx (hunk 2 above)
Comment (ematipico) [reply src/content/docs/en/guides/server-side-rendering.mdx]: The newline doesn't have any effect on the rendered markdown, I added a new line only to have the text in check without scrolling :)

**Review by sarah11918 — COMMENTED** [review]
Thanks for this, @ematipico ! I made some small editing suggestions for your consideration, so see what you think! And, one thing I'll point out is that for these docs pages, we don't arbitrarily break lines after some character amount if it's in the middle of a sentence.

I know some people like to start each new sentence on a new line. I personally find that distracting to edit when I need to also think about paragraph size and the flow of text, but I get why people like that pattern for easy targeting of each sentence and thats fine. But, we'd typically never break in the midlde of a sentence, and if you do that, I'm gonna change it!  :joy:

**Review by sarah11918 — APPROVED** [review]
This looks great to me @ematipico, thanks!  🎉 

Since I did suggest some text edits, we probably need a once-more over from @Yan-Thomas to be sure, but I'm approving for the feature relase.  :joy:

**Review by Yan-Thomas — APPROVED** [review]
LGTM!

File: src/content/docs/en/guides/server-side-rendering.mdx [hunk 5]
```diff
@@ -366,3 +367,32 @@ const factPromise = fetch('https://catfact.ninja/fact')
 ```
 
 In this example, `A name` will render while `personPromise` is loading. At that point, `A fact` will be appear, while `factPromise` is loading.
+
+
+### Split the SSR build into multiple files
+
+By default, Astro builds all pages into one single file called `entry.mjs`. You can configure [`split`](/en/reference/configuration-reference/#split) in your build options to instead emit a single file for each page:
```
Comment (ematipico) [inline src/content/docs/en/guides/server-side-rendering.mdx:374]:
```suggestion
By default, Astro builds all pages into one single file called `entry.mjs`. You can configure [`split`](/en/reference/configuration-reference/#buildsplit) in your build options to instead emit a single file for each page:
```

**Comment by ematipico** [comment]
@sarah11918 I just fixed the broken link. The CI should be OK now :)

**Comment by sarah11918** [comment]
Thank you! JJuggling too many things! lol... I'm merging as checks pass :laughing:

**Pull request merged**